kernel/examples/read-table-multi-threaded/src/main.rs

```diff
@@ -0,0 +1,286 @@
+    }
+
+    // have sent all scan files, drop this so threads will exit when there's no more work
+    drop(scan_file_tx);
+
+    // simply gather up each batch and print them
+    let batches: Vec<_> = record_batch_rx.iter().collect();
+    print_batches(&batches)?;
+    Ok(())
+}
+
+// this is the work each thread does
+fn do_work(
+    engine_interface: Arc<Box<dyn EngineInterface>>,
+    scan_state: GlobalScanState,
+    record_batch_tx: Sender<RecordBatch>,
+    scan_file_rx: spmc::Receiver<ScanFile>,
+) {
+    // get the type for the function calls
+    let engine_interface: &dyn EngineInterface = engine_interface.as_ref().as_ref();
+    let read_schema = Arc::new(scan_state.read_schema.clone());
+    loop {
+        // in a loop, try and get a ScanFile
+        match scan_file_rx.recv() {
+            Ok(scan_file) => {
+                // we got a scan file, let's process it
+                let root_url = Url::parse(&scan_state.table_root).unwrap();
+
+                // get the selection vector (i.e. deletion vector)
+                let mut selection_vector = scan_file
+                    .dv_info
+                    .get_selection_vector(engine_interface, &root_url)
+                    .unwrap();
+
+                // build the required metadata for our parquet handler to read this file
+                let location = root_url.join(&scan_file.path).unwrap();
+                let meta = FileMeta {
+                    last_modified: 0,
+                    size: scan_file.size as usize,
+                    location,
+                };
+
+                // this example uses the parquet_handler from the engine_client, but an engine could
+                // choose to use whatever method it might want to read a parquet file. The reader
+                // could, for example, fill in the parition columns, or apply deletion vectors. Here
+                // we assume a more naive parquet reader and fix the data up after the fact.
+                // further parallelism would also be possible here as we could read the parquet file
+                // in chunks where each thread reads one chunk. The engine would need to ensure
+                // enough meta-data was passed to each thread to correctly apply the selection
+                // vector
+                let read_results = engine_interface
+                    .get_parquet_handler()
+                    .read_parquet_files(&[meta], read_schema.clone(), None)
+                    .unwrap();
+
+                for read_result in read_results {
+                    let len = if let Ok(ref res) = read_result {
+                        res.length()
+                    } else {
+                        0
+                    };
+
+                    // ask the kernel to transform the physical data into the correct logical form
+                    let logical = transform_to_logical(
+                        engine_interface,
+                        read_result.unwrap(),
```

Comment:
Given that we anyway unconditionally `unwrap` the `read_result`, why not just unwrap it directly?
```suggestion
                    let read_result = read_result.unwrap();
                    let len = read_result.length()

                    // ask the kernel to transform the physical data into the correct logical form
                    let logical = transform_to_logical(
                        engine_interface,
                        read_result,
```